Streamline/restructure our “JavaScript implementations” list (#38623)
mdn/content: files/en-us/web/javascript/reference/javascript_technologies_overview/index.md

## JavaScript implementations

JavaScript engines used in current web browsers include:

- [SpiderMonkey](https://spidermonkey.dev/), used in Firefox (and its forks), Servo, and Flow (and in MongoDB, CouchDB, and more).
- [V8](https://v8.dev/), used in Chrome and in other Chromium-based browsers (and in [Node.js](https://nodejs.org/), [Deno](https://deno.com/), [Electron](https://www.electronjs.org/), and more).
- [JavaScriptCore](https://docs.webkit.org/Deep%20Dive/JSC/JavaScriptCore.html), used in Safari and in other WebKit-based browsers (and in [Bun](https://bun.sh/).
- [LibJS](https://serenityos.github.io/libjs-website/), used in [Ladybird](https://ladybird.org/).

Some JavaScript engines used in earlier browsers include:

> - [Carakan](<https://en.wikipedia.org/wiki/Presto_(browser_engine)#ECMAScript_engines>), used in Opera it became a Chromium-based browser.

Reviewer: ```suggestion
- [Carakan](<https://en.wikipedia.org/wiki/Presto_(browser_engine)#ECMAScript_engines>), used in Opera before it became a Chromium-based browser.
```

> - [Chakra](<https://en.wikipedia.org/wiki/Chakra_(JScript_engine)>), used in Internet Explorer and in Edge before it became a Chromium-based browser.

Reviewer: There are two unrelated Chakras.

```suggestion
- Microsoft's [Chakra](<https://en.wikipedia.org/wiki/Chakra_(JScript_engine)>) engine, used in Internet Explorer (although the language it implements is formally called "JScript" to avoid trademark issues). Earlier versions of Edge used a new JavaScript engine, confusingly also called [Chakra](<https://en.wikipedia.org/wiki/Chakra_(JavaScript_engine)>), before it became a Chromium-based browser.
```

Some JavaScript engines specifically tailored for non-browser purposes include:

- [Engine262](https://engine262.js.org/), written in JavaScript and intended essentially as a reference implementation of the language.

> - [Hermes](https://github.com/facebook/hermes), optimized for [React Native](https://reactnative.dev/docs/hermes).

Reviewer: ```suggestion
- Meta's [Hermes](https://github.com/facebook/hermes), optimized for [React Native](https://reactnative.dev/docs/hermes).
```

> - [Rhino](<https://en.wikipedia.org/wiki/Rhino_(JavaScript_engine)>), written in Java.
> - [GraalJS](https://www.graalvm.org/), written in Java and built on top of GraalVM.

Reviewer: ```suggestion
- Mozilla's [Rhino](<https://en.wikipedia.org/wiki/Rhino_(JavaScript_engine)>), written in Java.
- Oracle's [GraalJS](https://www.graalvm.org/), written in Java and built on top of GraalVM.
```
